lib/compat/wordpress-6.3/blocks.php

```diff
@@ -26,3 +26,93 @@ function gutenberg_add_selectors_property_to_block_type_settings( $settings, $me
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_add_selectors_property_to_block_type_settings', 10, 2 );
+
+/**
+ * Adds custom fields support to the wp_block post type so an unsynced option can be added and renames
+ * from Reusable block to Pattern.
+ *
+ * Note: This should be removed when the minimum required WP version is >= 6.3.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/51144
+ *
+ * @param array  $args Register post type args.
+ * @param string $post_type The post type string.
+ *
+ * @return array Register post type args.
+ */
+function gutenberg_add_custom_fields_to_wp_block( $args, $post_type ) {
+	if ( 'wp_block' === $post_type ) {
+		$args['labels']['name']                     = _x( 'Patterns', 'post type general name' );
+		$args['labels']['singular_name']            = _x( 'Pattern', 'post type singular name' );
+		$args['labels']['add_new_item']             = __( 'Add new Pattern' );
+		$args['labels']['new_item']                 = __( 'New Pattern' );
+		$args['labels']['edit_item']                = __( 'Edit Pattern' );
+		$args['labels']['view_item']                = __( 'View Pattern' );
+		$args['labels']['all_items']                = __( 'All Patterns' );
+		$args['labels']['search_items']             = __( 'Search Patterns' );
+		$args['labels']['not_found']                = __( 'No Patterns found.' );
+		$args['labels']['not_found_in_trash']       = __( 'No Patterns found in Trash.' );
+		$args['labels']['filter_items_list']        = __( 'Filter Patterns list' );
+		$args['labels']['items_list_navigation']    = __( 'Patterns list navigation' );
+		$args['labels']['items_list']               = __( 'Patterns list' );
+		$args['labels']['item_published']           = __( 'Pattern published.' );
+		$args['labels']['item_published_privately'] = __( 'Pattern published privately.' );
+		$args['labels']['item_reverted_to_draft']   = __( 'Pattern reverted to draft.' );
+		$args['labels']['item_scheduled']           = __( 'Pattern scheduled.' );
+		$args['labels']['item_updated']             = __( 'Pattern updated.' );
+		array_push( $args['supports'], 'custom-fields' );
```

Comment:
It might be worth separating the label changes from the addition of custom-fields support into two separate features so that the function name and doc block are appropriate.

Reply:
Done.

Reply:
If I'm not mistaken, this whole filter is not actually needed?